package proto_ff;

import "yd_fieldoptions.proto";


message E_MonsterMonster
{
	optional int64 m_monsterID = 1[(yd_fieldoptions.field_cname) = "怪物ID"];
	optional string m_monsterName = 2[(yd_fieldoptions.field_cname) = "怪物名", (yd_fieldoptions.field_bufsize) = 64];
	optional string m_mask = 3[(yd_fieldoptions.field_cname) = "掉落标记", (yd_fieldoptions.field_bufsize) = 64];
	optional int32 m_functionType = 4[(yd_fieldoptions.field_cname) = "功能类型"];
	optional int64 m_functionValue = 5[(yd_fieldoptions.field_cname) = "功能参数"];
	optional int32 m_taskType = 6[(yd_fieldoptions.field_cname) = "任务用怪类型"];
	optional int32 m_profession = 7[(yd_fieldoptions.field_cname) = "怪物职业"];
	optional int32 m_subProfession = 8[(yd_fieldoptions.field_cname) = "怪物子类型"];
	optional int32 m_damageAddType = 9[(yd_fieldoptions.field_cname) = "伤害加成类型"];
	optional int32 m_monsterLV = 10[(yd_fieldoptions.field_cname) = "怪物等级"];
	optional int32 m_droplimit = 11[(yd_fieldoptions.field_cname) = "最大掉落等级"];
	optional int64 m_displayID = 12[(yd_fieldoptions.field_cname) = "资源与表现ID"];
	optional int32 m_bossID = 13[(yd_fieldoptions.field_cname) = "首领ID"];
	optional int64 m_valuetype = 14[(yd_fieldoptions.field_cname) = "数值类型"];
	optional int32 m_valueParam = 15[(yd_fieldoptions.field_cname) = "数值参数"];
	optional int64 m_aiId = 16[(yd_fieldoptions.field_cname) = "AI表ID"];
	optional int32 m_hpSectionNum = 17[(yd_fieldoptions.field_cname) = "怪物血条段数"];
	optional int32 m_breedProb = 18[(yd_fieldoptions.field_cname) = "死亡触发新怪物几率"];
	optional int64 m_breedMonID = 19[(yd_fieldoptions.field_cname) = "触发生成怪物ID"];
	optional int32 m_breedMonLive = 20[(yd_fieldoptions.field_cname) = "怪物存活时间"];
	optional string m_itemCost = 21[(yd_fieldoptions.field_cname) = "道具消耗", (yd_fieldoptions.field_bufsize) = 64];
	optional int32 m_isBornActionOn = 22[(yd_fieldoptions.field_cname) = "是否启用出生动作"];
	optional int32 m_belongshow = 23[(yd_fieldoptions.field_cname) = "怪物掉落归属是否显示"];
	optional int32 m_dropGroupID = 24[(yd_fieldoptions.field_cname) = "掉落ID"];
	optional int32 m_harmdrop_type = 25[(yd_fieldoptions.field_cname) = "伤害计算方式"];
	optional int32 m_joninper = 26[(yd_fieldoptions.field_cname) = "参与奖最低百分比"];
	optional int32 m_btnRange = 27[(yd_fieldoptions.field_cname) = "按钮范围"];
	optional int64 m_btnId = 28[(yd_fieldoptions.field_cname) = "按钮ID"];
	optional int32 m_drivePoint = 29[(yd_fieldoptions.field_cname) = "怒气"];
	optional int32 m_corpseID = 30[(yd_fieldoptions.field_cname) = "尸体ID"];
	optional int64 m_battle = 31[(yd_fieldoptions.field_cname) = "战力"];
	optional int32 m_lvSupPve = 32[(yd_fieldoptions.field_cname) = "等级压制"];
	optional int32 m_powerSupPve = 33[(yd_fieldoptions.field_cname) = "战力压制"];
}


message Sheet_MonsterMonster
{
	repeated E_MonsterMonster E_MonsterMonster_List = 1[(yd_fieldoptions.field_arysize)=2048];
}

message E_MonsterDisplaySkillDesc
{
	optional int64 m_id = 1[(yd_fieldoptions.field_cname) = "id"];
}

message E_MonsterDisplay
{
	optional int64 m_displayID = 1[(yd_fieldoptions.field_cname) = "资源与表现ID"];
	optional int32 m_modelOffsetsX = 2[(yd_fieldoptions.field_cname) = "UI模型X坐标偏移"];
	optional int32 m_modelOffsetsY = 3[(yd_fieldoptions.field_cname) = "UI模型Y坐标偏移"];
	optional int32 m_modelScale = 4[(yd_fieldoptions.field_cname) = "UI模型缩放比例"];
	optional int32 m_modelRadius = 5[(yd_fieldoptions.field_cname) = "怪物模型半径"];
	optional int32 m_pathRadius = 6[(yd_fieldoptions.field_cname) = "寻路半径"];
	optional int32 m_collisionRadius = 7[(yd_fieldoptions.field_cname) = "寻路半径"];
	optional int32 m_rSpeed = 8[(yd_fieldoptions.field_cname) = "跑步速度"];
	repeated E_MonsterDisplaySkillDesc m_skill = 9[(yd_fieldoptions.field_cname) = "技能", (yd_fieldoptions.field_arysize) = 7];
}


message Sheet_MonsterDisplay
{
	repeated E_MonsterDisplay E_MonsterDisplay_List = 1[(yd_fieldoptions.field_arysize)=256];
}

message E_MonsterValue
{
	optional int64 m_ID = 1[(yd_fieldoptions.field_cname) = "数值ID"];
	optional int32 m_type = 2[(yd_fieldoptions.field_cname) = "怪物类型"];
	optional int32 m_monsterLV = 3[(yd_fieldoptions.field_cname) = "怪物等级"];
	optional int64 m_max_Hp = 4[(yd_fieldoptions.field_cname) = "最大生命"];
	optional int64 m_atk = 5[(yd_fieldoptions.field_cname) = "攻击"];
	optional int64 m_def = 6[(yd_fieldoptions.field_cname) = "防御"];
	optional int64 m_brk_Armor = 7[(yd_fieldoptions.field_cname) = "破甲"];
	optional int64 m_five_Dam = 8[(yd_fieldoptions.field_cname) = "五行伤害"];
	optional int64 m_five_Armor = 9[(yd_fieldoptions.field_cname) = "五行护甲"];
	optional int64 m_thump_Rate = 10[(yd_fieldoptions.field_cname) = "重击几率"];
	optional int64 m_thump_Resist = 11[(yd_fieldoptions.field_cname) = "重击抵抗"];
	optional int64 m_sky = 12[(yd_fieldoptions.field_cname) = "天道"];
	optional int64 m_sky_Dam = 13[(yd_fieldoptions.field_cname) = "天道伤害"];
	optional int64 m_sky_Redu_Resist = 14[(yd_fieldoptions.field_cname) = "天道减抗"];
	optional int64 m_sky_Resist = 15[(yd_fieldoptions.field_cname) = "天道抗性"];
	optional int64 m_kill_God = 16[(yd_fieldoptions.field_cname) = "弑神"];
	optional int64 m_god_Body = 17[(yd_fieldoptions.field_cname) = "神体"];
	optional int64 m_mortal_Rate = 18[(yd_fieldoptions.field_cname) = "凡免几率"];
	optional int64 m_mortal_Resist = 19[(yd_fieldoptions.field_cname) = "凡免抗性"];
	optional int64 m_fairy_Rate = 20[(yd_fieldoptions.field_cname) = "仙免几率"];
	optional int64 m_fairy_Resist = 21[(yd_fieldoptions.field_cname) = "仙免抗性"];
	optional int64 m_god_Rate = 22[(yd_fieldoptions.field_cname) = "神免几率"];
	optional int64 m_god_Resist = 23[(yd_fieldoptions.field_cname) = "神免抗性"];
	optional int64 m_crit_Rate = 24[(yd_fieldoptions.field_cname) = "暴击几率"];
	optional int64 m_crit_resit_Rate = 25[(yd_fieldoptions.field_cname) = "抗暴几率"];
	optional int64 m_crit_Dam = 26[(yd_fieldoptions.field_cname) = "暴击伤害"];
	optional int64 m_crit_Redu = 27[(yd_fieldoptions.field_cname) = "暴击减免"];
	optional int64 m_batter_Rate = 28[(yd_fieldoptions.field_cname) = "连击几率"];
	optional int64 m_batter_Resist = 29[(yd_fieldoptions.field_cname) = "连击抵抗"];
	optional int64 m_batter_Dam = 30[(yd_fieldoptions.field_cname) = "连击伤害"];
	optional int64 m_batter_Redu = 31[(yd_fieldoptions.field_cname) = "连击减免"];
	optional int64 m_brkdown_Rate = 32[(yd_fieldoptions.field_cname) = "击穿几率"];
	optional int64 m_brkdown_Resist = 33[(yd_fieldoptions.field_cname) = "击穿抵抗"];
	optional int64 m_dam_Add = 34[(yd_fieldoptions.field_cname) = "伤害加成"];
	optional int64 m_dam_Redu = 35[(yd_fieldoptions.field_cname) = "伤害减免"];
	optional int64 m_all_Add = 36[(yd_fieldoptions.field_cname) = "全能增伤"];
	optional int64 m_all_Redu = 37[(yd_fieldoptions.field_cname) = "全能减免"];
	optional int64 m_skill_Add = 38[(yd_fieldoptions.field_cname) = "技能增伤"];
	optional int64 m_skill_Redu = 39[(yd_fieldoptions.field_cname) = "技能减伤"];
	optional int64 m_rebound_Dam = 40[(yd_fieldoptions.field_cname) = "反弹伤害"];
	optional int64 m_hp_Steal = 41[(yd_fieldoptions.field_cname) = "生命窃取"];
	optional int64 m_dodge_Rate = 42[(yd_fieldoptions.field_cname) = "闪避几率"];
	optional int64 m_hit_Rate = 43[(yd_fieldoptions.field_cname) = "命中几率"];
	optional int64 m_blk_Rate = 44[(yd_fieldoptions.field_cname) = "格挡几率"];
	optional int64 m_brk_blk_Rate = 45[(yd_fieldoptions.field_cname) = "破档几率"];
	optional int64 m_blk_Free = 46[(yd_fieldoptions.field_cname) = "格挡免伤"];
	optional int64 m_blk_Pierce = 47[(yd_fieldoptions.field_cname) = "格挡穿透"];
	optional int64 m_rel_Armor = 48[(yd_fieldoptions.field_cname) = "真实护甲"];
	optional int64 m_rel_Dam = 49[(yd_fieldoptions.field_cname) = "真实伤害"];
	optional int64 m_five_Resist = 50[(yd_fieldoptions.field_cname) = "五行抗性"];
	optional int64 m_five_resist_Redu = 51[(yd_fieldoptions.field_cname) = "五行减抗"];
	optional int64 m_fire_Add = 52[(yd_fieldoptions.field_cname) = "离火增伤"];
	optional int64 m_fire_Redu = 53[(yd_fieldoptions.field_cname) = "离火减伤"];
	optional int64 m_ice_Add = 54[(yd_fieldoptions.field_cname) = "玄冰增伤"];
	optional int64 m_ice_Redu = 55[(yd_fieldoptions.field_cname) = "玄冰减伤"];
	optional int64 m_thunder_Add = 56[(yd_fieldoptions.field_cname) = "雷罚增伤"];
	optional int64 m_thunder_Rdu = 57[(yd_fieldoptions.field_cname) = "雷罚减伤"];
	optional int64 m_player_Add = 58[(yd_fieldoptions.field_cname) = "玩家减伤"];
	optional int64 m_player_Redu = 59[(yd_fieldoptions.field_cname) = "玩家增伤"];
	optional int64 m_leader_Add = 60[(yd_fieldoptions.field_cname) = "首领减伤"];
	optional int64 m_leader_Redu = 61[(yd_fieldoptions.field_cname) = "首领增伤"];
	optional int64 m_mon_Add = 62[(yd_fieldoptions.field_cname) = "怪物减伤"];
	optional int64 m_mon_Redu = 63[(yd_fieldoptions.field_cname) = "怪物增伤"];
	optional int64 m_var_Redu = 64[(yd_fieldoptions.field_cname) = "变身减伤"];
	optional int64 m_var_Add = 65[(yd_fieldoptions.field_cname) = "变身增伤"];
	optional int64 m_unusual_Resist = 66[(yd_fieldoptions.field_cname) = "异常抵抗"];
	optional int64 m_unusual_Hit = 67[(yd_fieldoptions.field_cname) = "异常命中"];
	optional int64 m_hp_Recover = 68[(yd_fieldoptions.field_cname) = "生命恢复"];
	optional int64 m_unusual_Add = 69[(yd_fieldoptions.field_cname) = "异常增伤"];
	optional int64 m_fix_Recover = 70[(yd_fieldoptions.field_cname) = "固定回复"];
	optional int64 m_high_hp_Add = 71[(yd_fieldoptions.field_cname) = "高血增伤"];
	optional int64 m_recover_hp_Add = 72[(yd_fieldoptions.field_cname) = "回血增效"];
	optional int64 m_week_Add = 73[(yd_fieldoptions.field_cname) = "虚弱增伤"];
	optional int64 m_shield = 74[(yd_fieldoptions.field_cname) = "护盾"];
	optional int64 m_exp = 75[(yd_fieldoptions.field_cname) = "怪物经验"];
}


message Sheet_MonsterValue
{
	repeated E_MonsterValue E_MonsterValue_List = 1[(yd_fieldoptions.field_arysize)=8192];
}

message E_MonsterValueparam
{
	optional int32 m_id = 1[(yd_fieldoptions.field_cname) = "id"];
	optional int32 m_P_max_Hp = 2[(yd_fieldoptions.field_cname) = "最大生命"];
	optional int32 m_P_atk = 3[(yd_fieldoptions.field_cname) = "攻击"];
	optional int32 m_P_def = 4[(yd_fieldoptions.field_cname) = "防御"];
	optional int32 m_P_brk_Armor = 5[(yd_fieldoptions.field_cname) = "破甲"];
	optional int32 m_P_five_Dam = 6[(yd_fieldoptions.field_cname) = "五行伤害"];
	optional int32 m_P_five_Armor = 7[(yd_fieldoptions.field_cname) = "五行护甲"];
	optional int32 m_P_thump_Rate = 8[(yd_fieldoptions.field_cname) = "重击几率"];
	optional int32 m_P_thump_Resist = 9[(yd_fieldoptions.field_cname) = "重击抵抗"];
	optional int32 m_P_sky = 10[(yd_fieldoptions.field_cname) = "天道"];
	optional int32 m_P_sky_Dam = 11[(yd_fieldoptions.field_cname) = "天道伤害"];
	optional int32 m_P_sky_Redu_Resist = 12[(yd_fieldoptions.field_cname) = "天道减抗"];
	optional int32 m_P_sky_Resist = 13[(yd_fieldoptions.field_cname) = "天道抗性"];
	optional int32 m_P_kill_God = 14[(yd_fieldoptions.field_cname) = "弑神"];
	optional int32 m_P_god_Body = 15[(yd_fieldoptions.field_cname) = "神体"];
	optional int32 m_P_mortal_Rate = 16[(yd_fieldoptions.field_cname) = "凡免几率"];
	optional int32 m_P_mortal_Resist = 17[(yd_fieldoptions.field_cname) = "凡免抗性"];
	optional int32 m_P_fairy_Rate = 18[(yd_fieldoptions.field_cname) = "仙免几率"];
	optional int32 m_P_fairy_Resist = 19[(yd_fieldoptions.field_cname) = "仙免抗性"];
	optional int32 m_P_god_Rate = 20[(yd_fieldoptions.field_cname) = "神免几率"];
	optional int32 m_P_god_Resist = 21[(yd_fieldoptions.field_cname) = "神免抗性"];
	optional int32 m_P_crit_Rate = 22[(yd_fieldoptions.field_cname) = "暴击几率"];
	optional int32 m_P_crit_resit_Rate = 23[(yd_fieldoptions.field_cname) = "抗暴几率"];
	optional int32 m_P_crit_Dam = 24[(yd_fieldoptions.field_cname) = "暴击伤害"];
	optional int32 m_P_crit_Redu = 25[(yd_fieldoptions.field_cname) = "暴击减免"];
	optional int32 m_P_batter_Rate = 26[(yd_fieldoptions.field_cname) = "连击几率"];
	optional int32 m_P_batter_Resist = 27[(yd_fieldoptions.field_cname) = "连击抵抗"];
	optional int32 m_P_batter_Dam = 28[(yd_fieldoptions.field_cname) = "连击伤害"];
	optional int32 m_P_batter_Redu = 29[(yd_fieldoptions.field_cname) = "连击减免"];
	optional int32 m_P_brkdown_Rate = 30[(yd_fieldoptions.field_cname) = "击穿几率"];
	optional int32 m_P_brkdown_Resist = 31[(yd_fieldoptions.field_cname) = "击穿抵抗"];
	optional int32 m_P_dam_Add = 32[(yd_fieldoptions.field_cname) = "伤害加成"];
	optional int32 m_P_dam_Redu = 33[(yd_fieldoptions.field_cname) = "伤害减免"];
	optional int32 m_P_all_Add = 34[(yd_fieldoptions.field_cname) = "全能增伤"];
	optional int32 m_P_all_Redu = 35[(yd_fieldoptions.field_cname) = "全能减免"];
	optional int32 m_P_skill_Add = 36[(yd_fieldoptions.field_cname) = "技能增伤"];
	optional int32 m_P_skill_Redu = 37[(yd_fieldoptions.field_cname) = "技能减伤"];
	optional int32 m_P_rebound_Dam = 38[(yd_fieldoptions.field_cname) = "反弹伤害"];
	optional int32 m_P_hp_Steal = 39[(yd_fieldoptions.field_cname) = "生命窃取"];
	optional int32 m_P_dodge_Rate = 40[(yd_fieldoptions.field_cname) = "闪避几率"];
	optional int32 m_P_hit_Rate = 41[(yd_fieldoptions.field_cname) = "命中几率"];
	optional int32 m_P_blk_Rate = 42[(yd_fieldoptions.field_cname) = "格挡几率"];
	optional int32 m_P_brk_blk_Rate = 43[(yd_fieldoptions.field_cname) = "破档几率"];
	optional int32 m_P_blk_Free = 44[(yd_fieldoptions.field_cname) = "格挡免伤"];
	optional int32 m_P_blk_Pierce = 45[(yd_fieldoptions.field_cname) = "格挡穿透"];
	optional int32 m_P_rel_Armor = 46[(yd_fieldoptions.field_cname) = "真实护甲"];
	optional int32 m_P_rel_Dam = 47[(yd_fieldoptions.field_cname) = "真实伤害"];
	optional int32 m_P_five_Resist = 48[(yd_fieldoptions.field_cname) = "五行抗性"];
	optional int32 m_P_five_resist_Redu = 49[(yd_fieldoptions.field_cname) = "五行减抗"];
	optional int32 m_P_fire_Add = 50[(yd_fieldoptions.field_cname) = "离火增伤"];
	optional int32 m_P_fire_Redu = 51[(yd_fieldoptions.field_cname) = "离火减伤"];
	optional int32 m_P_ice_Add = 52[(yd_fieldoptions.field_cname) = "玄冰增伤"];
	optional int32 m_P_ice_Redu = 53[(yd_fieldoptions.field_cname) = "玄冰减伤"];
	optional int32 m_P_thunder_Add = 54[(yd_fieldoptions.field_cname) = "雷罚增伤"];
	optional int32 m_P_thunder_Rdu = 55[(yd_fieldoptions.field_cname) = "雷罚减伤"];
	optional int32 m_P_player_Add = 56[(yd_fieldoptions.field_cname) = "玩家减伤"];
	optional int32 m_P_player_Redu = 57[(yd_fieldoptions.field_cname) = "玩家增伤"];
	optional int32 m_P_leader_Add = 58[(yd_fieldoptions.field_cname) = "首领减伤"];
	optional int32 m_P_leader_Redu = 59[(yd_fieldoptions.field_cname) = "首领增伤"];
	optional int32 m_P_mon_Add = 60[(yd_fieldoptions.field_cname) = "怪物减伤"];
	optional int32 m_P_mon_Redu = 61[(yd_fieldoptions.field_cname) = "怪物增伤"];
	optional int32 m_P_var_Redu = 62[(yd_fieldoptions.field_cname) = "变身减伤"];
	optional int32 m_P_var_Add = 63[(yd_fieldoptions.field_cname) = "变身增伤"];
	optional int32 m_P_unusual_Resist = 64[(yd_fieldoptions.field_cname) = "异常抵抗"];
	optional int32 m_P_unusual_Hit = 65[(yd_fieldoptions.field_cname) = "异常命中"];
	optional int32 m_P_hp_Recover = 66[(yd_fieldoptions.field_cname) = "生命恢复"];
	optional int32 m_P_unusual_Add = 67[(yd_fieldoptions.field_cname) = "异常增伤"];
	optional int32 m_P_fix_Recover = 68[(yd_fieldoptions.field_cname) = "固定回复"];
	optional int32 m_P_high_hp_Add = 69[(yd_fieldoptions.field_cname) = "高血增伤"];
	optional int32 m_P_recover_hp_Add = 70[(yd_fieldoptions.field_cname) = "回血增效"];
	optional int32 m_P_week_Add = 71[(yd_fieldoptions.field_cname) = "虚弱增伤"];
	optional int64 m_shield = 72[(yd_fieldoptions.field_cname) = "护盾"];
	optional int64 m_exp = 73[(yd_fieldoptions.field_cname) = "怪物经验"];
}


message Sheet_MonsterValueparam
{
	repeated E_MonsterValueparam E_MonsterValueparam_List = 1[(yd_fieldoptions.field_arysize)=128];
}

message E_MonsterDrop
{
	optional int32 m_id = 1[(yd_fieldoptions.field_cname) = "掉落包ID"];
	optional int32 m_group = 2[(yd_fieldoptions.field_cname) = "掉落组"];
	optional int32 m_own = 3[(yd_fieldoptions.field_cname) = "归属权"];
	optional int32 m_type = 4[(yd_fieldoptions.field_cname) = "掉落方式"];
	repeated int64 m_boxID = 5[(yd_fieldoptions.field_cname) = "box", (yd_fieldoptions.field_arysize) = 5];
}


message Sheet_MonsterDrop
{
	repeated E_MonsterDrop E_MonsterDrop_List = 1[(yd_fieldoptions.field_arysize)=512];
}

message E_MonsterCollect
{
	optional int32 m_ID = 1[(yd_fieldoptions.field_cname) = "类型"];
	optional int32 m_type = 2[(yd_fieldoptions.field_cname) = "采集物类型"];
	optional int32 m_num = 3[(yd_fieldoptions.field_cname) = "公共采集次数"];
	optional int32 m_breakDown = 4[(yd_fieldoptions.field_cname) = "是否能被打断"];
	optional int32 m_dropGroupID = 5[(yd_fieldoptions.field_cname) = "采集获得"];
	optional int32 m_time = 6[(yd_fieldoptions.field_cname) = "读条时间"];
	optional int32 m_getType = 7[(yd_fieldoptions.field_cname) = "获得技能的方式"];
	optional int32 m_skillTime = 8[(yd_fieldoptions.field_cname) = "技能间隔"];
	optional int32 m_skillID = 9[(yd_fieldoptions.field_cname) = "采集获得技能"];
	optional int32 m_dupID = 10[(yd_fieldoptions.field_cname) = "副本ID"];
	optional int32 m_monsteID = 11[(yd_fieldoptions.field_cname) = "怪物ID"];
}


message Sheet_MonsterCollect
{
	repeated E_MonsterCollect E_MonsterCollect_List = 1[(yd_fieldoptions.field_arysize)=32];
}
